File: java/lib/src/main/java/com/svix/models/StreamPortalAccessIn.java
// This file is @generated
package com.svix.models;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.svix.Utils;

import lombok.EqualsAndHashCode;
import lombok.ToString;

import java.util.LinkedHashSet;
import java.util.Set;

@ToString
@EqualsAndHashCode
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonAutoDetect(getterVisibility = Visibility.NONE, setterVisibility = Visibility.NONE)
public class StreamPortalAccessIn {
    @JsonProperty private Long expiry;
    @JsonProperty private Set<String> featureFlags;
    @JsonProperty private String sessionId;

    public StreamPortalAccessIn() {}

    public StreamPortalAccessIn expiry(Long expiry) {
        this.expiry = expiry;
        return this;
    }

    /**
     * How long the token will be valid for, in seconds.
     *
     * <p>Valid values are between 1 hour and 7 days. The default is 7 days.
     *
     * @return expiry
     */
    @javax.annotation.Nullable
    public Long getExpiry() {
        return expiry;
    }

    public void setExpiry(Long expiry) {
        this.expiry = expiry;
    }

    public StreamPortalAccessIn featureFlags(Set<String> featureFlags) {
        this.featureFlags = featureFlags;
        return this;
    }

    public StreamPortalAccessIn addFeatureFlagsItem(String featureFlagsItem) {
        if (this.featureFlags == null) {
            this.featureFlags = new LinkedHashSet<>();
        }
        this.featureFlags.add(featureFlagsItem);

        return this;
    }

    /**
     * The set of feature flags the created token will have access to.
     *
     * @return featureFlags
     */
    @javax.annotation.Nullable
    public Set<String> getFeatureFlags() {
        return featureFlags;
    }

    public void setFeatureFlags(Set<String> featureFlags) {
        this.featureFlags = featureFlags;
    }

    public StreamPortalAccessIn sessionId(String sessionId) {
        this.sessionId = sessionId;
        return this;
    }

    /**
     * An optional session ID to attach to the token.
     *
     * <p>When expiring tokens with "Expire All", you can include the session ID to only expire
     * tokens that were created with that session ID.
     *
     * @return sessionId
     */
    @javax.annotation.Nullable
    public String getSessionId() {
        return sessionId;
    }

    public void setSessionId(String sessionId) {
        this.sessionId = sessionId;
    }

    /**
     * Create an instance of StreamPortalAccessIn given an JSON string
     *
     * @param jsonString JSON string
     * @return An instance of StreamPortalAccessIn
     * @throws JsonProcessingException if the JSON string is invalid with respect to
     *     StreamPortalAccessIn
     */
    public static StreamPortalAccessIn fromJson(String jsonString) throws JsonProcessingException {
        return Utils.getObjectMapper().readValue(jsonString, StreamPortalAccessIn.class);
    }

    /**
     * Convert an instance of StreamPortalAccessIn to an JSON string
     *
     * @return JSON string
     */
    public String toJson() throws JsonProcessingException {
        return Utils.getObjectMapper().writeValueAsString(this);
    }
}

File: php/src/Models/StreamPortalAccessIn.php
<?php

// this file is @generated
declare(strict_types=1);

namespace Svix\Models;

class StreamPortalAccessIn implements \JsonSerializable
{
    private array $setFields = [];

    /**
     * @param list<string>|null $featureFlags the set of feature flags the created token will have access to
     * @param int|null          $expiry       How long the token will be valid for, in seconds.
     *
     * Valid values are between 1 hour and 7 days. The default is 7 days.
     * @param string|null $sessionId An optional session ID to attach to the token.
     *
     * When expiring tokens with "Expire All", you can include the session ID to only expire tokens that were created with that session ID.
     */
    private function __construct(
        public readonly ?array $featureFlags = null,
        public readonly ?int $expiry = null,
        public readonly ?string $sessionId = null,
        array $setFields = [],
    ) {
        $this->setFields = $setFields;
    }

    /**
     * Create an instance of StreamPortalAccessIn with required fields.
     */
    public static function create(
    ): self {
        return new self(
            featureFlags: null,
            expiry: null,
            sessionId: null,
            setFields: []
        );
    }

    public function withFeatureFlags(?array $featureFlags): self
    {
        $setFields = $this->setFields;
        $setFields['featureFlags'] = true;

        return new self(
            featureFlags: $featureFlags,
            expiry: $this->expiry,
            sessionId: $this->sessionId,
            setFields: $setFields
        );
    }

    public function withExpiry(?int $expiry): self
    {
        $setFields = $this->setFields;
        $setFields['expiry'] = true;

        return new self(
            featureFlags: $this->featureFlags,
            expiry: $expiry,
            sessionId: $this->sessionId,
            setFields: $setFields
        );
    }

    public function withSessionId(?string $sessionId): self
    {
        $setFields = $this->setFields;
        $setFields['sessionId'] = true;

        return new self(
            featureFlags: $this->featureFlags,
            expiry: $this->expiry,
            sessionId: $sessionId,
            setFields: $setFields
        );
    }

    public function jsonSerialize(): mixed
    {
        $data = [
        ];

        if (null !== $this->featureFlags) {
            $data['featureFlags'] = $this->featureFlags;
        }
        if (isset($this->setFields['expiry'])) {
            $data['expiry'] = $this->expiry;
        }
        if (isset($this->setFields['sessionId'])) {
            $data['sessionId'] = $this->sessionId;
        }

        return \Svix\Utils::newStdClassIfArrayIsEmpty($data);
    }

    /**
     * Create an instance from a mixed obj.
     */
    public static function fromMixed(mixed $data): self
    {
        return new self(
            featureFlags: \Svix\Utils::getValFromJson($data, 'featureFlags', false, 'StreamPortalAccessIn'),
            expiry: \Svix\Utils::deserializeInt($data, 'expiry', false, 'StreamPortalAccessIn'),
            sessionId: \Svix\Utils::deserializeString($data, 'sessionId', false, 'StreamPortalAccessIn')
        );
    }

    /**
     * Create an instance from a json string.
     */
    public static function fromJson(string $json): self
    {
        $data = json_decode(json: $json, associative: true, depth: 512, flags: JSON_THROW_ON_ERROR);

        return self::fromMixed($data);
    }
}
